runtime/jvmti/jvmtiThreadGroup.c

```diff
@@ -180,23 +186,36 @@ jvmtiGetThreadGroupChildren(jvmtiEnv* env,
 			/* Construct the Children Groups array under a lock */
 
  			threadGroupObject = *((j9object_t*) group);
```

Comment:
This should move to the else case.

Reply:
This is required by following `threadGroupObject = (j9object_t) currentThread->javaVM->internalVMFunctions->objectMonitorEnter(currentThread, threadGroupObject)`.
Maybe you meant `L200` which can be moved into `else` case instead?

Reply:
I was just wrong.